net-mgmt/pfSense-pkg-Telegraf/files/usr/local/pkg/telegraf.inc

```diff
@@ -0,0 +1,115 @@
+<?php
+/*
+ * telegraf.inc
+ *
+ * part of pfSense (https://www.pfsense.org)
+ * Copyright (c) 2017 Rubicon Communications, LLC (Netgate)
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+require_once("functions.inc");
+require_once("globals.inc");
+require_once("pkg-utils.inc");
+require_once("service-utils.inc");
+require_once("util.inc");
+
+function telegraf_resync_config() {
+	global $g, $config;
+
+	if (is_array($config['installedpackages']['telegraf'])) {
+		$telegraf_conf = &$config['installedpackages']['telegraf']['config'][0];
+	} else {
+		$telegraf_conf = array();
+	}
+
+	conf_mount_rw();
+
+	/* disable telegraf if not enabled */
+	if ($telegraf_conf['enable'] != "on") {
+		if (is_service_running("telegraf")) {
+			stop_service("telegraf");
+		}
+		unlink_if_exists("/usr/local/etc/rc.d/telegraf.sh");
+		unlink_if_exists("/usr/local/etc/telegraf.conf");
+		return;
+	}
+
+	/* generate telegraf.conf */
+	$cfg = <<< EOD
+# This file is automatically generated by pfSense #
+[agent]
+	interval = "10s"
+	round_interval = true
+
+[[inputs.cpu]]
+	percpu = true
+	totalcpu = true
+	fielddrop = ["time_*"]
+
+[[inputs.disk]]
+	ignore_fs = ["tmpfs", "devtmpfs"]
+
+[[inputs.diskio]]
+
+[[inputs.kernel]]
+
+[[inputs.mem]]
+
+[[inputs.net]]
+
+[[inputs.processes]]
+
+[[inputs.swap]]
+
+[[inputs.system]]
+
+[[outputs.influxdb]]
+
+[[outputs.influxdb]]
+	urls = ["{$telegraf_conf['influx_server']}"]
+	database = "{$telegraf_conf['influx_db']}"
+
+EOD;
+
+	if (!empty($telegraf_conf['influx_user'])) {
+		$cfg .= "\tusername = \"" . $telegraf_conf['influx_user'] . "\"\n";
+	}
+	if (!empty($telegraf_conf['influx_pass'])) {
+		$cfg .= "\tpassword = \"" . $telegraf_conf['influx_pass'] . "\"\n";
+	}
+	$conffile = "/usr/local/etc/telegraf.conf";
+	file_put_contents($conffile, $cfg);
+
+	/* generate telegraf.sh rcfile */
+	$pidfile = "{$g['varrun_path']}/telegraf.pid";
+	$logfile = "{$g['varlog_path']}/telegraf.log";
+	write_rcfile(array(
+		"file" => "telegraf.sh",
+		"start" => "/usr/sbin/daemon -crP {$pidfile} /usr/local/bin/telegraf -config={$conffile} 2>> {$logfile}",
+		"stop" => "/bin/kill `/bin/cat {$pidfile}`"
```

Comment:
What is going to happen if telegraf.log become a huge file? There are no processes taking care of it. Doesn't it support syslog?